pkg/natdiscovery/remote_endpoint.go

```diff
@@ -86,17 +102,26 @@ func (rn *remoteEndpointNAT) sinceLastTransition() time.Duration {
 }
 
 func (rn *remoteEndpointNAT) hasTimedOut() bool {
-	return time.Since(rn.started) > toDuration(&totalTimeout)
+	return time.Since(rn.started) > rn.timeout
 }
 
 func (rn *remoteEndpointNAT) useLegacyNATSettings() {
-	rn.useNAT = rn.endpoint.Spec.NATEnabled
-	if rn.endpoint.Spec.NATEnabled {
+	switch {
+	case rn.usingLoadBalancer:
+		rn.useNAT = true
 		rn.useIP = rn.endpoint.Spec.PublicIP
 		rn.transitionToState(selectedPublicIP)
+		klog.V(log.DEBUG).Infof("using NAT for the load balancer backed endpoint %q, using public IP %q", rn.endpoint.Spec.CableName,
+			rn.useIP)
+
+	case rn.endpoint.Spec.NATEnabled:
+		rn.useIP = rn.endpoint.Spec.PublicIP
+		rn.useNAT = rn.endpoint.Spec.NATEnabled
+		rn.transitionToState(selectedPublicIP)
 		klog.V(log.DEBUG).Infof("using NAT legacy settings for endpoint %q, using public IP %q", rn.endpoint.Spec.CableName,
 			rn.useIP)
-	} else {
+
+	default:
```

Comment:
```suggestion
	default:
	    rn.useNAT = false
```

Reply:
Right, at this point it was working because it's false by default, but better let's be explicit.